core/src/main/java/org/fao/geonet/api/records/attachments/AbstractStore.java

```diff
@@ -226,11 +211,39 @@ public final MetadataResource putResource(ServiceContext context, String metadat
     @Override
     public final MetadataResource putResource(ServiceContext context, String metadataUuid, URL fileUrl,
             MetadataResourceVisibility visibility, Boolean approved) throws Exception {
-        String filename = getFilenameFromHeader(fileUrl);
-        if (filename == null) {
+
+        // Open a connection to the URL
+        HttpURLConnection connection = (HttpURLConnection) fileUrl.openConnection();
+        connection.setInstanceFollowRedirects(true);
+        connection.setRequestMethod("GET");
+
+        // Check if the response code is OK
+        int responseCode = connection.getResponseCode();
+        if (responseCode != HttpURLConnection.HTTP_OK) {
+            throw new IOException("Unexpected response code: " + responseCode);
+        }
+
+        // Extract filename from Content-Disposition header if present otherwise use the filename from the URL
+        String contentDisposition = connection.getHeaderField(HttpHeaders.CONTENT_DISPOSITION);
+        String filename = null;
+        if (contentDisposition != null) {
+            filename = ContentDisposition.parse(contentDisposition).getFilename();
+        }
+        if (filename == null || filename.isEmpty()) {
             filename = getFilenameFromUrl(fileUrl);
         }
-        return putResource(context, metadataUuid, filename, fileUrl.openStream(), null, visibility, approved);
+
+        // Check if the content length is within the allowed limit
+        long contentLength = connection.getContentLengthLong();
+        if (contentLength > maxUploadSize) {
+            throw new InputStreamLimitExceededException(maxUploadSize, contentLength);
+        }
+
+        // Put the resource but limit the input stream to the max upload size plus one byte
+        // so we can check if the file is larger than the allowed size
```

Comment:
This comment is a little confusion.  Maybe the following would be clearer?

```suggestion
        // Upload the resource while ensuring the input stream does not exceed the maximum allowed size.
```